flake.nix

```diff
@@ -15,14 +27,183 @@
+                           }
+                          function do-clang-format {
+                             banner "In place clang-format"
+                             cd $SRC_ROOT
+                             include_regex=".*\.(c|h)$"
+                             src_files=`find ./api -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./bin -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./crypto -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./stuffer -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./error -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tls -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./utils -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/unit -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+                             src_files+=" "
+                             src_files+=`find ./tests/testlib -name .git -prune -o -regextype posix-egrep -regex "$include_regex" -print`
+            		             echo $src_files | xargs -n 1 -P $(nproc) clang-format -style=file -i
+                           }
+          '';
+
+          packages = [
+            # Build Depends
+            openssl_1_1_1
+            pkgs.cmake
+            # Other Libcryptos
+            # openssl_0_9_8 openssl_1_0_2 openssl_3_0
+            # libressl pkgs.boringssl
+
+            # Integration Deps
+            pythonEnv
+            corretto-8
+            gnutls-3-7
+
+            # C Compiler Tooling: llvmPkgs.clangUseLLVM -- wrapper to overwrite default compiler with clang
+            llvmPkgs.llvm
+            llvmPkgs.llvm-manpages
+            llvmPkgs.libclang
+            llvmPkgs.clang-manpages
+
+            # Linters/Formatters
+            pkgs.shellcheck
+            pkgs.cppcheck
+            pkgs.nixfmt
+            pkgs.python39Packages.pep8
+
+            # Rust
+            # TODO: can we use the version in bindings/rust/rust-toolchain
+            # it goes against the spirit of nix to use rustup... but we might
+            # have to -- using a new rust is liable to get us in trouble.
+            pkgs.rustc
+            pkgs.cargo
```

Comment:
I _think_ I would prefer to use rustup since the `rust-toolchain` file makes the versioning just work everywhere.

But one issue you can run into with rustup-installed toolchains is nix store paths get out of date on upgrades and you have to uninstall the toolchain and reinstall it.

Managing the rust versions here, we definitely want to be pinning to MSRV so people don't accidentally develop features that are only compatible with stable versions.